073-decouple-ca-management-from-end-entity-certificate-management.md

```diff
@@ -0,0 +1,250 @@
+3. At the beginning of the reconcile loop of components, the cluster operator generates the Cluster Operator certificate and stores it in the <CLUSTER_NAME>-cluster-operator-certs Kubernetes Secret
+   It does this even though the state is UNTRUSTED, because the Kubernetes Secret was missing
+4. The component reconcilers (KafkaReconciler, ZooKeeperReconciler, EntityOperatorReconciler etc):
+   1. generate their certificates, store them in Kubernetes Secrets
+   2. create their pods annotated with the certificates Kubernetes Secret and new shared Kubernetes Secret mounted onto the pod
+5. At the end of the reconcile loop of components, the cluster operator updates the shared Kubernetes Secret to change the state of the CA certificate to TRUSTED_IN_USE
+
+When the CA private key is updated (either because it has expired or the user has incremented the generation to indicate they've manually updated the Kubernetes Secret):
+
+1. If Strimzi is managing the CA, the CaReconciler renews the Cluster and Client CAs certificates and keys. The old certificate and key can be removed from the Kubernetes Secret.
+2. The CaReconciler takes the Cluster CA certificate and places it in the new shared Kubernetes Secret with the state UNTRUSTED. It also updates the old CA certificate to have the state PHASE_OUT.
+3. The component reconcilers (KafkaReconciler, ZooKeeperReconciler, EntityOperatorReconciler etc) see the new UNTRUSTED CA certificate and:
+   1. Roll the pods to trust the new CA certificate
+4. At the end of the reconcile loop the cluster operator updates the shared Kubernetes Secret to change the state of the CA certificate to TRUSTED_UNUSED
+5. This change triggers a new reconcile loop
+6. At the beginning of the reconcile loop the cluster operator sees there is a CA certificate in the state TRUSTED_UNUSED and regenerates the Cluster Operator certificate, signed by the new CA key and stores it in the <CLUSTER_NAME>-cluster-operator-certs Kubernetes Secret
+7. The component reconcilers (KafkaReconciler, ZooKeeperReconciler, EntityOperatorReconciler etc) see the new TRUSTED_UNUSED CA certificate and:
+   1. Generate new certificates signed by the new CA key
+   2. Roll the pods to use the new certificates
+8. At the end of the reconcile loop the cluster operator updates the shared Kubernetes Secret to change the state of the CA certificate to TRUSTED_IN_USE and the old CA certificate to ZOMBIE, indicating that no end entity certificates are using the CA certificate
+9. In a future loop the CaReconciler removes the old CA certificate from the shared Kubernetes Secret (because it is observed to be in the `ZOMBIE` state).
+
+Notes:
+The existing annotations strimzi.io/ca-cert-generation and strimzi.io/ca-key-generation can still be used by the CaReconciler and the cluster operator to track which certificates the Kubernetes Pods are trusting/using.
```

Comment:
This got me completely lost as it is now not clear how to the generations relate to your state secret.